[Bug]: Linting CI check fails on all PRs due to pre-existing errors

### What happened?

# Bug Description

  The linting CI workflow is failing for all PRs due to pre-existing linting errors in files unrelated to the changes. This blocks valid PRs from
  being merged even when the contributor's code has zero linting issues.

  ## Current

  When a contributor opens a PR with clean code:

  1. ✅ Their changes have no linting errors
  2. ❌ CI fails because of pre-existing errors in other files
  3. ❌ PR appears "broken" even though it's valid

  ## CI Error Output

  ```bash
  cd litellm
  poetry run ruff check .

  integrations/arize/_utils.py:90:5: PLR0915 Too many statements (91 > 50)
  integrations/langfuse/langfuse_otel.py:90:9: PLR0915 Too many statements (61 > 50)
  llms/vertex_ai/ocr/transformation.py:14:42: F401 imported but unused
  proxy/guardrails/guardrail_hooks/noma/noma.py:12:53: F401 imported but unused
  proxy/guardrails/guardrail_hooks/noma/noma.py:26:59: F401 imported but unused
  router.py:775:9: PLR0915 Too many statements (56 > 50)
```
  Found 13 errors.
  Error: Process completed with exit code 1.

  Files

  - integrations/arize/_utils.py - PLR0915 (function too long), F401 (unused imports)
  - integrations/langfuse/langfuse_otel.py - PLR0915 (function too long)
  - integrations/langfuse/langfuse_otel_attributes.py - F401 (unused imports)
  - llms/vertex_ai/ocr/transformation.py - F401 (unused import)
  - proxy/guardrails/guardrail_hooks/noma/noma.py - F401 (unused imports)
  - router.py - PLR0915 (function too long)

  Impact

  - Blocks valid contributions: Contributors with clean code see their PRs fail CI
  - Bad developer experience: New contributors might think they did something wrong
  - Maintainer overhead: Maintainers need to manually verify which errors are pre-existing

  Proposed Solutions

  Option 1: Fix the pre-existing errors (Recommended)
```bash
  cd litellm
  poetry run ruff check . --fix  # Fixes auto-fixable errors (F401 imports)
```
  Then manually refactor functions that are too long (PLR0915).

  Option 2: Only lint changed files in CI

  Modify CI workflow to only check files modified in the PR:
```yaml
  - name: Get changed files
    id: changed-files
    uses: tj-actions/changed-files@v39
    with:
      files: |
        litellm/**/*.py

  - name: Run Ruff on changed files
    if: steps.changed-files.outputs.any_changed == 'true'
    run: |
      cd litellm
      poetry run ruff check ${{ steps.changed-files.outputs.all_changed_files }}
```
  Option 3: Add ruff baseline/ignore

  Create .ruff.toml with baseline ignores for pre-existing errors until they're fixed.

##Details

  CI linting should:
  - ✅ Fail if the contributor's changes have linting errors
  - ✅ Pass if the contributor's changes are clean, even if other files have pre-existing errors


### Relevant log output

```shell

```

### Are you a ML Ops Team?

No

### What LiteLLM version are you on ?

1.79.1

### Twitter / LinkedIn details

_No response_

--- a/litellm/caching/redis_cache.py
+++ b/litellm/caching/redis_cache.py
@@ -1071,9 +1071,9 @@ class RedisCache(BaseCache):
             
             # Test the connection
             ping_result = await redis_client.ping()
-            
+
             # Close the connection
-            await redis_client.aclose()
+            await redis_client.aclose()  # type: ignore[attr-defined]
             
             if ping_result:
                 return {

--- a/litellm/caching/redis_cluster_cache.py
+++ b/litellm/caching/redis_cluster_cache.py
@@ -83,10 +83,10 @@ class RedisClusterCache(RedisCache):
             )
             
             # Test the connection
-            ping_result = await redis_client.ping()
-            
+            ping_result = await redis_client.ping()  # type: ignore[attr-defined]
+
             # Close the connection
-            await redis_client.aclose()
+            await redis_client.aclose()  # type: ignore[attr-defined]
             
             if ping_result:
                 return {

--- a/litellm/completion_extras/litellm_responses_transformation/transformation.py
+++ b/litellm/completion_extras/litellm_responses_transformation/transformation.py
@@ -541,7 +541,7 @@ class LiteLLMResponsesTransformationHandler(CompletionTransformationBridge):
     def _map_reasoning_effort(self, reasoning_effort: Union[str, Dict[str, Any]]) -> Optional[Reasoning]:
         # If dict is passed, convert it directly to Reasoning object
         if isinstance(reasoning_effort, dict):
-            return Reasoning(**reasoning_effort)
+            return Reasoning(**reasoning_effort)  # type: ignore[typeddict-item]
 
         # If string is passed, map without summary (default)
         if reasoning_effort == "high":

--- a/litellm/litellm_core_utils/litellm_logging.py
+++ b/litellm/litellm_core_utils/litellm_logging.py
@@ -1,7 +1,6 @@
 # What is this?
 ## Common Utility file for Logging handler
 # Logging function -> log the exact model details + what's being sent | Non-Blocking
-import asyncio
 import copy
 import datetime
 import json

--- a/litellm/responses/utils.py
+++ b/litellm/responses/utils.py
@@ -381,7 +381,7 @@ class ResponseAPILoggingUtils:
                 cached_tokens=response_api_usage.input_tokens_details.cached_tokens,
                 audio_tokens=response_api_usage.input_tokens_details.audio_tokens,
             )
-        usage = Usage(
+        usage = Usage(  # type: ignore[assignment]
             prompt_tokens=prompt_tokens,
             completion_tokens=completion_tokens,
             total_tokens=prompt_tokens + completion_tokens,
@@ -392,4 +392,4 @@ class ResponseAPILoggingUtils:
         if hasattr(response_api_usage, "cost") and response_api_usage.cost is not None:
             setattr(usage, "cost", response_api_usage.cost)
 
-        return usage
+        return usage  # type: ignore[return-value]

--- a/litellm/responses/utils.py
+++ b/litellm/responses/utils.py
@@ -361,17 +361,19 @@ class ResponseAPILoggingUtils:
 
     @staticmethod
     def _transform_response_api_usage_to_chat_usage(
-        usage: Optional[Union[dict, ResponseAPIUsage]],
+        usage_input: Optional[Union[dict, ResponseAPIUsage]],
     ) -> Usage:
         """Tranforms the ResponseAPIUsage object to a Usage object"""
-        if usage is None:
+        if usage_input is None:
             return Usage(
                 prompt_tokens=0,
                 completion_tokens=0,
                 total_tokens=0,
             )
         response_api_usage: ResponseAPIUsage = (
-            ResponseAPIUsage(**usage) if isinstance(usage, dict) else usage
+            ResponseAPIUsage(**usage_input)
+            if isinstance(usage_input, dict)
+            else usage_input
         )
         prompt_tokens: int = response_api_usage.input_tokens or 0
         completion_tokens: int = response_api_usage.output_tokens or 0
@@ -381,7 +383,7 @@ class ResponseAPILoggingUtils:
                 cached_tokens=response_api_usage.input_tokens_details.cached_tokens,
                 audio_tokens=response_api_usage.input_tokens_details.audio_tokens,
             )
-        usage = Usage(  # type: ignore[assignment]
+        chat_usage = Usage(
             prompt_tokens=prompt_tokens,
             completion_tokens=completion_tokens,
             total_tokens=prompt_tokens + completion_tokens,
@@ -390,6 +392,6 @@ class ResponseAPILoggingUtils:
 
         # Preserve cost attribute if it exists on ResponseAPIUsage
         if hasattr(response_api_usage, "cost") and response_api_usage.cost is not None:
-            setattr(usage, "cost", response_api_usage.cost)
+            setattr(chat_usage, "cost", response_api_usage.cost)
 
-        return usage  # type: ignore[return-value]
+        return chat_usage

--- a/litellm/secret_managers/aws_secret_manager_v2.py
+++ b/litellm/secret_managers/aws_secret_manager_v2.py
@@ -238,7 +238,7 @@ class AWSSecretsManagerV2(BaseAWSLLM, BaseSecretManager):
                 tags_list = tags
             else:
                 raise ValueError("Tags must be a dict or list of {Key, Value} pairs")
-            data["Tags"] = tags_list
+            data["Tags"] = tags_list  # type: ignore[assignment]
 
         endpoint_url, headers, body = self._prepare_request(
             action="CreateSecret",

